tests/Unit/NumericalAlgorithms/Interpolation/Test_IrregularInterpolant.cpp

```diff
@@ -0,0 +1,177 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <memory>
+#include <random>
+#include <string>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/IndexIterator.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "NumericalAlgorithms/Interpolation/IrregularInterpolant.hpp"
+#include "NumericalAlgorithms/Spectral/LegendreGaussLobatto.hpp"
+#include "PointwiseFunctions/MathFunctions/MathFunction.hpp"// IWYU pragma: keep
+#include "PointwiseFunctions/MathFunctions/PowX.hpp" // IWYU pragma: keep
+#include "PointwiseFunctions/MathFunctions/TensorProduct.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Literals.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+#include "tests/Utilities/MakeWithRandomValues.hpp"
+// IWYU pragma: no_forward_declare MathFunction
+// IWYU pragma: no_forward_declare PowX
+// IWYU pragma: no_forward_declare Tensor
+
+namespace TestTags {
+
+template <size_t Dim>
+struct Vector : db::SimpleTag {
+  using type = tnsr::I<DataVector, Dim>;
+  static std::string name() noexcept {return "Vector"; }
+  static auto fill_values(const MathFunctions::TensorProduct<Dim>& f,
+                          const tnsr::I<DataVector, Dim>& x) noexcept {
+    auto result = make_with_value<tnsr::I<DataVector, Dim>>(x, 0.);
+    const auto f_of_x = f(x);
+    for (size_t d = 0; d < Dim; ++d) {
+      result.get(d) = (d + 0.5) * get(f_of_x);
+    }
+    return result;
+  }
+};
+
+template <size_t Dim>
+struct SymmetricTensor : db::SimpleTag {
+  using type = tnsr::ii<DataVector, Dim>;
+  static std::string name() noexcept {return "SymmetricTensor"; }
+  static auto fill_values(const MathFunctions::TensorProduct<Dim>& f,
+                          const tnsr::I<DataVector, Dim>& x) noexcept {
+    auto result = make_with_value<tnsr::ii<DataVector, Dim>>(x, 0.);
+    const auto f_of_x = f(x);
+    for (size_t i = 0; i < Dim; ++i) {
+      for (size_t j = i; j < Dim; ++j) {  // Symmetry
+        result.get(i, j) = (i + j + 0.33) * get(f_of_x);
+      }
+    }
+    return result;
+  }
+};
+
+}  // namespace TestTags
+
+namespace {
+
+template <size_t Dim>
+void test_interpolate_to_points(const Index<Dim>& mesh) noexcept {
+  // Fill target interpolation coordinates with random values
+  std::random_device r;
+  const auto seed = r();
+  std::mt19937 generator(seed);
+  INFO("seed" << seed);
+  std::uniform_real_distribution<> dist(-1., 1.);
```

Comment:
Do we require the target points to be in `[-1, 1)` or could we make this `[-1.2, 3.2)` (or something else not so symmetry)?

Reply:
Will and I think we should keep the random values in the range actually used, i.e. [-1,1]